src/v/transform/rpc/client.cc

```diff
@@ -807,6 +808,42 @@ client::generate_remote_report(model::node_id node) {
     co_return std::move(resp).value().report;
 }
 
+std::optional<cluster::topic_configuration>
+client::find_topic_cfg(model::topic_namespace_view tp_ns) const {
+    return _topic_metadata->find_topic_cfg(tp_ns);
+}
+
+ss::future<cluster::errc> client::create_transform_logs_topic() {
+    co_return co_await retry(
+      [this]() { return try_create_transform_logs_topic(); });
+}
+
+ss::future<cluster::errc> client::try_create_transform_logs_topic() {
+    cluster::topic_properties topic_props;
+    topic_props.batch_max_bytes
+      = config::shard_local_cfg().kafka_batch_max_bytes();
+    topic_props.retention_bytes = tristate<size_t>(-1);
```

Comment:
@rockwotj @mattschumpert  - Per discussion, we should come up with a sensible default for this.

Reply:
for retention overall, or just bytes?

Reply:
bytes.  Someone could easily flood disks with logs and upset the apple cart of local retention, space management etc. for everyone using the cluster. capping the logs to a total byte size thats a small fraction of the disk seems prudent. not sure what that number should be.

Reply:
Something similar to our max broker log size seems reasonable

Reply:
>something similar to our max broker log size

Makes sense to me 👍

Reply:
Do we ship with a default syslog config for this? I can't seem to find anything

Reply:
Set to 2_GiB per https://github.com/redpanda-data/redpanda/pull/16485#issuecomment-1934631531